### ISSUE TITLE
napi/transform-react: expose recoverable React Compiler diagnostics

### ISSUE THREAD

**Issue by jasonkuhrt** [oxc-project/oxc#26318]
### What

`transformSync` from `oxc-transform-react` gives no way to learn **why** the React Compiler skipped a component. `TransformResult.errors` is fatal-only in 0.148.0 — recoverable diagnostics are dropped on success, and `panicThreshold` defaults to `'none'`, so every ordinary bailout is invisible by construction.

0.147.0 did surface them (as warnings in `errors`); 0.148.0 removed that.

### Why it matters

Adopting the React Compiler on an existing codebase is a burndown: you need an inventory of which components fail to compile **and why**, so the fixable ones (rule suppressions, conditional hooks, unpreservable memoization) can be separated from compiler limitations you can only wait on.

Without reasons, the only available signal is inspecting the emitted code for the memo cache (`const $ = _c(...)`), which tells you *that* a function was skipped and nothing more. Babel's plugin exposed this via its `logger` option, so this is a capability regression for anyone migrating off it.

Pinning 0.147.0 as a diagnostic-only side channel works, but it means running a second, older compiler over the whole codebase purely to read the messages.

### Ask

Expose recoverable diagnostics on the NAPI result — a `diagnostics` array alongside `errors`, or an opt-in `logger`/`events` option mirroring Babel's. The Rust `oxc_react_compiler` side already carries category and reason; only the binding drops them.

Severity and source span would make it directly actionable; the message string alone would already be a large improvement.

**Comment by JacktheRanger** [comment]
+1 from another consumer that lost its only observation channel in 0.148.0, with three angles the issue doesn't cover yet.

**1. It makes the #25392 fix unobservable.** Since 0.145.0 (#25394) the transform honors `eslint-disable-next-line react-hooks/*` comments — silently, as Babel does. Our repo has no ESLint anymore, so those comments have exactly one job: opting a component out of the compiler. We keep a small gate that asks the compiler which suppressions actually took effect (the `react-compiler(Suppression)` diagnostic's label is precise to the comment) and fails when one is misspelled, mis-placed, or has drifted away from the function it was written for. On 0.148.0 the suppression still takes effect but nothing reports it, so a typo'd suppression is indistinguishable from a working one — fail-open in the direction that silently changes memo semantics.

**2. `outputMode: 'lint'` now reports nothing.** Its docstring (and `noEmit`'s: "Analyze and report diagnostics without applying compiler output") promises diagnostics, but on 0.148.0 it returns `errors: []` as well — the README's "omitted in every output mode" makes lint mode a no-op at the Node boundary.

**3. `panicThreshold` is not a workaround.** Escalating recoverable diagnostics to fatal stops at the first one: `fatal: true`, no code, and later bailouts in the same file are never enumerated (verified with two suppressed functions in one file → `errors.length === 1`, label on the first comment).

Minimal repro on 0.148.0 — the `_c(` count shows the suppression is honored, `errors` never says so:

```js
import { transformSync } from 'oxc-transform-react';
const code = `import { useEffect, useState } from 'react';
export function A({ x }) {
  const [s, setS] = useState(0);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => { setS(x); }, []);
  return <div>{s}</div>;
}
export function B({ y }) { const [t] = useState(y); return <span>{t}</span>; }`;
for (const [label, extra] of [['default', {}], ['no suppressions', { eslintSuppressionRules: [] }], ['lint', { outputMode: 'lint' }], ['panic', { panicThreshold: 'all_errors' }]]) {
  const r = transformSync('a.tsx', code, { lang: 'tsx', jsx: 'preserve', reactCompiler: { target: '19', ...extra } });
  console.log(label.padEnd(16), { fatal: r.fatal, errors: r.errors.length, memoized: (r.code.match(/_c\(/g) ?? []).length });
}
// default          { fatal: false, errors: 0, memoized: 1 }
// no suppressions  { fatal: false, errors: 0, memoized: 2 }
// lint             { fatal: false, errors: 0, memoized: 0 }
// panic            { fatal: true,  errors: 1, memoized: 0 }
```

Any of the shapes proposed above would work for us; a `diagnostics` array next to `errors` is the smallest API surface, and the Rust side already has category, span and help text for each one — the binding currently discards `diagnostics` from `CompileResult::Success` on the way out. One note for others landing here: the changelog lists #26128 under Bug Fixes, so Dependabot minor groups pull it in without any breaking-change flag; a test was the only thing that caught it on our side.
